src/server/game/Spells/SpellInfo.cpp

```diff
@@ -1131,6 +1131,14 @@ bool SpellInfo::IsAllowingDeadTarget() const
     return HasAttribute(SPELL_ATTR2_CAN_TARGET_DEAD) || Targets & (TARGET_FLAG_CORPSE_ALLY | TARGET_FLAG_CORPSE_ENEMY | TARGET_FLAG_UNIT_DEAD);
 }
 
+bool SpellInfo::IsRaidMemberTarget() const
+{
+    for (uint8 i = 0; i < MAX_SPELL_EFFECTS; ++i)
+        if (Effects[i].IsEffect() && (Effects[i].TargetA.GetTarget() & (TARGET_UNIT_CASTER_AREA_RAID | TARGET_UNIT_TARGET_RAID | TARGET_UNIT_TARGET_AREA_RAID_CLASS)))
+            return true;
+    return false;
+}
+
```

Comment:
@boxa forgive me for being so blunt but in my opinion spell targeting and aura stacking rules in mangos are complete garbage with the amount of hacks "exclusions" it needs for various things

On topic: you can try using SpellImplicitTargetInfo::GetCheckType - TARGET_CHECK_PARTY, TARGET_CHECK_RAID, TARGET_CHECK_RAID_CLASS,